### PR TITLE
Services branch

### PR DESCRIPTION
**Pull request by Divo123-tech**

### CODE REVIEW

**Comment by Divo123-tech** [comment]
changes warehouseId to string because warehouse model hasn't been fixed yet, will change later

File: src/service/item.service.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            return true;
+        }
+    }
+    return false; // Return false if kode doesn't exist or if k is not provided
+}
+
+
+const getAll = async() : Promise<Array<ItemInterface>> => {
+    try {
+        return await itemModel.find({isActive: true});
```
Comment (yharsono) [inline src/service/item.service.ts:16]: will this return _id? cause _id is not part of the ItemInterface?

File: src/controller/items.controller.ts [hunk 2]
```diff
@@ -1,5 +1,39 @@
...
+        console.log(req.body)
+      res.status(200).json(await itemService.add(req.body));
+    } catch (err: any) {
+      res.status(400).json(err);
+    }
+  }
+
+  static async delete(req: Request, res: Response): Promise<void> {
+    try {
+      res.status(200).json(await itemService.remove(req.body.id));
```
Comment (yharsono) [inline src/controller/items.controller.ts:24]: this should be part of params, not body. 
the URL should be: DELETE  /items/:itemId

File: src/controller/items.controller.ts [hunk 3]
```diff
@@ -1,5 +1,39 @@
-import TemplateController from './template.controller';
-import itemModel, { ItemInterface } from '../models/items.model'; // Import itemModel and IItem interface
+import itemService from "../service/item.service";
+import { Request, Response } from 'express';
+
+class ItemController {
+  static async getAll(req: Request, res: Response): Promise<void> {
+    try {
+      res.status(200).json(await itemService.getAll());
```
Comment (yharsono) [inline src/controller/items.controller.ts:7]: nit: for readability, its better to do:
```
const items = await itemService.getAll()
res.status(200).json(items)
```

File: src/service/item.service.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        // return newItem;
+    }
+    catch (err: any) {
+        throw new Error(err)
+    }
+}
+
+const remove = async(id : string) : Promise<Boolean> =>{
+    try{
+        await itemModel.findByIdAndUpdate(id, {isActive: false}, {new:true})
```
Comment (yharsono) [inline src/service/item.service.ts:35]: why do you need to set {new:true} if the result is not actually being used?

File: src/models/items.model.ts [hunk 5]
```diff
@@ -49,11 +49,15 @@ const itemSchema: Schema<ItemInterface> = new Schema<ItemInterface>({
...
+    //     ref: 'warehouse',
+    //     required: true,
+    //     default: '',
+    // },
     warehouseId: {
-        type: mongoose.Schema.Types.ObjectId,
-        ref: 'warehouse',
-        required: true,
-        default: '',
+        type: String,
```
Comment (yharsono) [inline src/models/items.model.ts:59]: why is this not ObjectId?

**Review by yharsono — CHANGES_REQUESTED** [review]

**Comment by Divo123-tech** [comment]
<h3>Removed DoesKodeExist Function</h3>
used .save to update or edit the function which now directly activates the validation making the DoesKodeExist function unnecessary. 

<h3>Added a modified Item Interface for the Front End </h3>
created an interface which has all the properties of the item interface without isActive, and used Object.assign to ensure that isActive cannot be modified from true unless its using the delete route.

<h3>Changed delete route</h3>
delete route now has id in the params not the body

**Pull request merged**